File: learn_android/code/MyToDoList1/app/src/main/java/com/hkc/mytodolist1/LoginActivity.java
package com.hkc.mytodolist1;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.databinding.DataBindingUtil;

import android.app.ProgressDialog;
import android.content.Intent;
import android.os.Bundle;
import android.text.TextUtils;
import android.view.View;
import android.view.WindowManager;
import android.widget.Toast;

import com.google.android.gms.tasks.OnCompleteListener;
import com.google.android.gms.tasks.Task;
import com.google.firebase.auth.AuthResult;
import com.google.firebase.auth.FirebaseAuth;
import com.hkc.mytodolist1.databinding.ActivityLoginBinding;


public class LoginActivity extends AppCompatActivity {
    ActivityLoginBinding binding;

    private FirebaseAuth mAuth;
    private ProgressDialog loader;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        getWindow().setFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN,WindowManager.LayoutParams.FLAG_FULLSCREEN);
//        setContentView(R.layout.activity_login);
        binding = DataBindingUtil.setContentView(this,R.layout.activity_login);

        mAuth = FirebaseAuth.getInstance();
        loader = new ProgressDialog(this);

        setSupportActionBar(binding.toolbarlogin);
        getSupportActionBar().setTitle("LOGIN");



        binding.loginPageQuestion.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(LoginActivity.this,RegistrationActivity.class);
                startActivity(intent);
            }
        });

        binding.buttonLogin.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                String email = binding.edittextLoginemail.getText().toString().trim();
                String password = binding.edittextLoginpassword.getText().toString().trim();

                if (TextUtils.isEmpty(email)){
                    binding.edittextLoginemail.setError("請輸入Email");
                    return;
                }
                if (TextUtils.isEmpty(password)){
                    binding.edittextLoginpassword.setError("請輸入Password");
                    return;
                }else{
                    loader.setMessage("Login in progress");
                    loader.setCanceledOnTouchOutside(false);
                    loader.show();

                    mAuth.signInWithEmailAndPassword(email,password).addOnCompleteListener(new OnCompleteListener<AuthResult>() {
                        @Override
                        public void onComplete(@NonNull Task<AuthResult> task) {
                            if (task.isSuccessful()){
                                Intent intent = new Intent(LoginActivity.this,HomeActivity.class);
                                startActivity(intent);
                                finish();
                                loader.dismiss();
                            }else{
                                String error = task.getException().toString();
                                Toast.makeText(LoginActivity.this,"Login Failed" + error ,Toast.LENGTH_LONG).show();
                                loader.dismiss();
                            }
                        }
                    });
                }
            }
        });


    }
}

File: learn_android/code/demo2/Atm1/app/src/main/java/com/hkc/atm/LoginActivity.kt
package com.hkc.atm

import android.app.Activity
import android.content.Context
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Toast
import androidx.appcompat.app.AlertDialog
import com.hkc.atm.databinding.ActivityLoginBinding

class LoginActivity : AppCompatActivity() {
    private lateinit var binding: ActivityLoginBinding


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
//        setContentView(R.layout.activity_login)

        binding = ActivityLoginBinding.inflate(layoutInflater)
        setContentView(binding.root)

        val userid = getSharedPreferences("atm", MODE_PRIVATE)
            .getString("PREF_USERID","")
        binding.edUserid.setText(userid)

        binding.button.setOnClickListener {
            login()
        }



        fun cancel(view:View){



        }



    }

    private fun login (){
        val userid = binding.edUserid.text.toString()
        val passwd = binding.edPasswd.text.toString()
        if (userid == "jack" && passwd == "1234"){
            getSharedPreferences("atm", Context.MODE_PRIVATE)
                .edit()
                .putString("PREF_USERID",userid)
                .apply()

            Toast.makeText(this,"登入成功",Toast.LENGTH_LONG).show()
            intent.putExtra("LOGIN_USERID",userid)
            intent.putExtra("LOGIN_PASSWD",passwd)
            setResult(Activity.RESULT_OK, intent)

            finish()
        }else{
            AlertDialog.Builder(this)
                .setTitle("ATM")
                .setMessage("登入失敗")
                .setPositiveButton("OK",null)
                .show()
        }


    }

}